app/components/sy-toggle/template.hbs

```diff
@@ -5,6 +5,11 @@
   tabindex="0"
   role="button"
   {{on "click" (optional (unless @disabled (fn (optional @onToggle))))}}
+  {{on-key
+    "Space"
+    (optional (unless @disabled (fn (optional @onToggle))))
```

Comment:
Why not just?
```suggestion
    @onToggle
```

Reply:
To me it seems like calling the component `SyToggle` without the `@onToggle` doesn't make much sense. And if it does, there is an `optional` to much in that statement.

Reply:
The first one covers the case, that `@disabled` is `true` and the second `optional` is needed if `@onToggle` is undefined.
Normally one would use the constructor to assert `this.args.onToggle` is not empty, right?

Reply:
> The first one covers the case, that @disabled is true and the second optional is needed if @onToggle is undefined.

I don't think `fn` is needed and then you only need an `optional` for the `{{on`.
> Normally one would use the constructor to assert this.args.onToggle is not empty, right?

Yep with `import { assert } from '@ember/debug';`

Reply:
Reworked. Plz check :)